compiler/noirc_evaluator/src/ssa/interpreter/intrinsics.rs

```diff
@@ -690,3 +772,73 @@ fn new_embedded_curve_point(
         ],
     ))
 }
+
+/// Convert a [Value] to a vector of [FieldElement] for printing.
+fn value_to_fields(value: &Value) -> Vec<FieldElement> {
+    fn go(value: &Value, fields: &mut Vec<FieldElement>) {
+        match value {
+            Value::Numeric(numeric_value) => fields.push(numeric_value.convert_to_field()),
+            Value::Reference(reference_value) => {
+                if let Some(value) = reference_value.element.borrow().as_ref() {
+                    go(value, fields);
+                }
+            }
+            Value::ArrayOrSlice(array_value) => {
+                for value in array_value.elements.borrow().iter() {
+                    go(value, fields);
+                }
+            }
+            Value::Function(id) => {
+                fields.push(FieldElement::from(id.to_u32()));
+                // We should push the fields in the `env` of the function, but I'm not sure how to get them.
+            }
+            Value::Intrinsic(_) => {
+                todo!("convert intrinsic to fields")
+            }
+            Value::ForeignFunction(_) => {
+                todo!("convert foreign function to fields")
```

Comment:
Should we capture issues for these?

Reply:
I wouldn't know what to do with them to be honest, like, I don't even know if they can come up.

Reply:
Makes sense. I think its fine to leave this like that then, if it eventually comes up (unlikely!) we can fix it.